lib/src/config.dart

```diff
@@ -4,11 +4,51 @@
 
 library linter.src.config;
 
+import 'package:analyzer/plugin/options.dart';
 import 'package:yaml/yaml.dart';
 
+/// Process the given option [fileContents] and produce a corresponding
+/// [LintConfig].
+LintConfig processAnalysisOptionsFile(String fileContents, {String fileUrl}) {
+  var yaml = loadYamlNode(fileContents, sourceUrl: fileUrl);
+  if (yaml is YamlMap) {
+    return _parseConfig(yaml);
+  }
+  return null;
+}
+
+LintConfig _parseConfig(Map optionsMap) {
+  if (optionsMap != null) {
+    var options = optionsMap['linter'];
```

Comment:
I'm not sure that we should assume that all plugins will be written as packages, or that if they are they will be published on pub. But it does seem reasonable to say that plugins that meet those criteria should use the package name and that others should use a fully qualified name.